specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+
+  const loading = React.useContext(LoadingContext);
+
+  const [saveBlocked, setSavedBlocked] = React.useState(false);
+
+  const [isEditing, _handleEditing, _disableEditing, toggleIsEditing] =
+    useBooleanState();
+
+  const fields = React.useMemo(
+    () => model.literalFields.filter((field) => !field.isVirtual),
+    [model]
+  );
+
+  const relationships = React.useMemo(
+    () =>
+      model.relationships.filter(
+        (relationship) =>
+          !(['one-to-many', 'many-to-many'] as RA<RelationshipType>).includes(
+            relationship.type
+          ) && !relationship.isVirtual
+      ),
+    [model]
+  );
+
+  React.useEffect(() => {
+    setSavedBlocked(
+      tableRules
+        .filter(({ duplicates }) => duplicates !== undefined)
+        .some(({ duplicates }) => duplicates!.totalDuplicates > 0)
+    );
+  }, [tableRules]);
+
+  const handleRuleValidation = React.useCallback(
+    (newRule: UniquenessRule, index: number) => {
+      const filteredRule: UniquenessRule = {
+        ...newRule,
+        fields: newRule.fields.filter(
+          (field, index) => newRule.fields.indexOf(field) === index
+        ),
+        scopes: newRule.scopes.filter(
+          (scope, index) => newRule.scopes.indexOf(scope) === index
+        ),
+      };
+      loading(
+        validateUniqueness(
+          container.name as keyof Tables,
+          filteredRule.fields as unknown as RA<never>,
+          filteredRule.scopes as unknown as RA<never>
+        ).then((duplicates) => {
+          const isNewRule = index > tableRules.length;
+          setTableRules((previous) =>
+            isNewRule
+              ? [...previous!, { rule: filteredRule, duplicates }]
+              : replaceItem(tableRules, index, {
+                  rule: filteredRule,
+                  duplicates,
+                })
+          );
+
+          return filteredRule;
+        })
+      );
+    },
+    [container.name, loading, tableRules, setTableRules]
+  );
+
+  return (
+    <Dialog
+      buttons={
+        <>
+          <Button.DialogClose>{commonText.close()}</Button.DialogClose>
+          {saveBlocked ? (
+            <Button.Danger
+              className="cursor-not-allowed"
+              onClick={(): void => undefined}
+            >
+              {icons.exclamation}
+              {commonText.save()}
+            </Button.Danger>
+          ) : (
+            <Submit.Save
+              disabled={tableRules === uniquenessRules[model.name]}
+              onClick={(): void => {
+                loading(
+                  ajax(
+                    `/businessrules/uniqueness_rules/${schema.domainLevelIds.discipline}/`,
+                    {
+                      // eslint-disable-next-line @typescript-eslint/naming-convention
+                      headers: { Accept: 'application/json' },
+                      method: tableRules.some(({ rule }) => rule.id === null)
+                        ? 'POST'
+                        : 'PUT',
```

Comment:
do we need separate POST and PUT endpoints?

the back-end code appears almost identical, with just a slightly different permission check and returned status code:

```py
        if request.method == 'POST':
            check_permission_targets(
                request.specify_collection.id, request.specify_user.id, [SetUniqueRulePT.create])
        elif request.method == 'PUT':
            check_permission_targets(
                request.specify_collection.id, request.specify_user.id, [SetUniqueRulePT.update])
```

so the client gets to decide which permission it wants to have checked? 🤣

and, you don't even have the POST request type documented in the opeanapi schema